parquet-avro/src/test/java/org/apache/parquet/avro/TestAvroSchemaConverter.java

```diff
@@ -105,6 +138,61 @@ public void testAllTypes() throws Exception {
     Schema schema = new Schema.Parser().parse(
         Resources.getResource("all.avsc").openStream());
     testAvroToParquetConversion(
+        NEW_BEHAVIOR, schema,
+        "message org.apache.parquet.avro.myrecord {\n" +
+            // Avro nulls are not encoded, unless they are null unions
+            "  required boolean myboolean;\n" +
+            "  required int32 myint;\n" +
+            "  required int64 mylong;\n" +
+            "  required float myfloat;\n" +
+            "  required double mydouble;\n" +
+            "  required binary mybytes;\n" +
+            "  required binary mystring (UTF8);\n" +
+            "  required group mynestedrecord {\n" +
+            "    required int32 mynestedint;\n" +
+            "  }\n" +
+            "  required binary myenum (ENUM);\n" +
+            "  required group myarray (LIST) {\n" +
+            "    repeated group list {\n" +
+            "      required int32 element;\n" +
+            "    }\n" +
+            "  }\n" +
+            "  required group myemptyarray (LIST) {\n" +
+            "    repeated group list {\n" +
+            "      required int32 element;\n" +
+            "    }\n" +
+            "  }\n" +
+            "  optional group myoptionalarray (LIST) {\n" +
+            "    repeated group list {\n" +
+            "      required int32 element;\n" +
+            "    }\n" +
+            "  }\n" +
+            "  required group myarrayofoptional (LIST) {\n" +
+            "    repeated group list {\n" +
+            "      optional int32 element;\n" +
+            "    }\n" +
+            "  }\n" +
+            "  required group mymap (MAP) {\n" +
+            "    repeated group map (MAP_KEY_VALUE) {\n" +
+            "      required binary key (UTF8);\n" +
+            "      required int32 value;\n" +
+            "    }\n" +
+            "  }\n" +
+            "  required group myemptymap (MAP) {\n" +
+            "    repeated group map (MAP_KEY_VALUE) {\n" +
+            "      required binary key (UTF8);\n" +
+            "      required int32 value;\n" +
```

Comment:
I'm fine with that, but how about as a separate issue? This doesn't change the structure of maps.